### ISSUE TITLE
Arbitrary attributes of function objects

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#1181]
**Describe the bug**
Undefined attributes of a function object have type `Any` and do not produce an error on access.  This masks accidentally attempting to access attributes or calling methods on the result type without having previously called the function.

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That Pyright would report that `foo` has no attribute `startswith`.

**Screenshots or Code**
```python
# pyright: strict

def foo() -> str:
    ...

foo.startswith('bar')  # No error
```

**VS Code extension or command-line**
Either, 1.1.86

**Comment by erictraut** [comment]
I've been avoiding adding this check since the earliest days of pyright because I knew it would produce many errors in my team's Python code base. It has been a "TODO" comment in the pyright code for more than a year.

I've implemented it as a new diagnostic rule called "reportFunctionMemberAccess". When typeCheckingMode is set to "basic", it is off by default because it produces too many false positives in most code bases. When typeCheckingMode is set to "strict" it is enabled. As with all diagnostic rules, its severity can be specified independently of other rules.

**Comment by erictraut** [comment]
This is now addressed in pyright 1.1.87, which I just published. It will also be included in the next release of pylance.
